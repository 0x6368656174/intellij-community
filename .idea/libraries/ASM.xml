<component name="libraryTable">
  <library name="ASM" type="repository">
    <properties include-transitive-deps="false" jar-repository-id="intellij-dependencies" maven-id="org.jetbrains.intellij.deps:asm-all:9.5.2">
      <verification>
        <artifact url="file://$MAVEN_REPOSITORY$/org/jetbrains/intellij/deps/asm-all/9.5.2/asm-all-9.5.2.jar">
          <sha256sum>ce77969b9f5d910e3c269a4898ecc03ae90b02c99b4194781df779e2f93515e4</sha256sum>
        </artifact>
      </verification>
    </properties>
    <CLASSES>
      <root url="jar://$MAVEN_REPOSITORY$/org/jetbrains/intellij/deps/asm-all/9.5.2/asm-all-9.5.2.jar!/" />
    </CLASSES>
    <JAVADOC />
    <SOURCES>
      <root url="jar://$MAVEN_REPOSITORY$/org/jetbrains/intellij/deps/asm-all/9.5.2/asm-all-9.5.2-sources.jar!/" />
    </SOURCES>
  </library>
</component>