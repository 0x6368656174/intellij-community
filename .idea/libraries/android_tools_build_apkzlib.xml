<component name="libraryTable">
  <library name="android.tools.build.apkzlib" type="repository">
    <properties maven-id="com.android.tools.build:apkzlib:7.3.1" />
    <CLASSES>
      <root url="jar://$MAVEN_REPOSITORY$/com/android/tools/build/apkzlib/7.3.1/apkzlib-7.3.1.jar!/" />
      <root url="jar://$MAVEN_REPOSITORY$/com/android/tools/build/apksig/7.3.1/apksig-7.3.1.jar!/" />
      <root url="jar://$MAVEN_REPOSITORY$/org/bouncycastle/bcprov-jdk15on/1.67/bcprov-jdk15on-1.67.jar!/" />
      <root url="jar://$MAVEN_REPOSITORY$/org/bouncycastle/bcpkix-jdk15on/1.67/bcpkix-jdk15on-1.67.jar!/" />
      <root url="jar://$MAVEN_REPOSITORY$/com/google/guava/guava/30.1-jre/guava-30.1-jre.jar!/" />
    </CLASSES>
    <JAVADOC />
    <SOURCES />
  </library>
</component>