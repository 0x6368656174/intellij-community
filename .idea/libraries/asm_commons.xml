<component name="libraryTable">
  <library name="asm-commons" type="repository">
    <properties maven-id="org.ow2.asm:asm-commons:9.5" />
    <CLASSES>
      <root url="jar://$MAVEN_REPOSITORY$/org/ow2/asm/asm-commons/9.5/asm-commons-9.5.jar!/" />
      <root url="jar://$MAVEN_REPOSITORY$/org/ow2/asm/asm/9.5/asm-9.5.jar!/" />
      <root url="jar://$MAVEN_REPOSITORY$/org/ow2/asm/asm-tree/9.5/asm-tree-9.5.jar!/" />
    </CLASSES>
    <JAVADOC />
    <SOURCES />
  </library>
</component>