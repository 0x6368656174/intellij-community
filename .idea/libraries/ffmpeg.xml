<component name="libraryTable">
  <library name="ffmpeg" type="repository">
    <properties maven-id="org.bytedeco:ffmpeg:5.1.2-1.5.8" />
    <CLASSES>
      <root url="jar://$MAVEN_REPOSITORY$/org/bytedeco/ffmpeg/5.1.2-1.5.8/ffmpeg-5.1.2-1.5.8.jar!/" />
      <root url="jar://$MAVEN_REPOSITORY$/org/bytedeco/javacpp/1.5.8/javacpp-1.5.8.jar!/" />
    </CLASSES>
    <JAVADOC>
      <root url="http://bytedeco.org/javacpp-presets/ffmpeg/apidocs" />
      <root url="http://bytedeco.org/javacpp/apidocs" />
    </JAVADOC>
    <SOURCES />
  </library>
</component>