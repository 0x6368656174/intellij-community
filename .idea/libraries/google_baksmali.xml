<component name="libraryTable">
  <library name="google-baksmali" type="repository">
    <properties maven-id="com.android.tools.smali:smali-baksmali:3.0.0">
      <exclude>
        <dependency maven-id="org.smali:dexlib2"/>
        <dependency maven-id="com.google.code.findbugs:jsr305"/>
        <dependency maven-id="com.google.guava:guava"/>
        <dependency maven-id="com.beust:jcommander"/>
      </exclude>
    </properties>
    <CLASSES>
      <root url="jar://$MAVEN_REPOSITORY$/com/android/tools/smali/smali-baksmali/3.0.0/smali-baksmali-3.0.0.jar!/"/>
      <root url="jar://$MAVEN_REPOSITORY$/com/android/tools/smali/smali-util/3.0.0/smali-util-3.0.0.jar!/"/>
    </CLASSES>
    <JAVADOC/>
    <SOURCES/>
  </library>
</component>