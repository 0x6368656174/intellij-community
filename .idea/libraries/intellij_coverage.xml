<component name="libraryTable">
  <library name="intellij-coverage" type="repository">
    <properties include-transitive-deps="false" maven-id="org.jetbrains.intellij.deps:intellij-coverage-agent:1.0.731">
      <verification>
        <artifact url="file://$MAVEN_REPOSITORY$/org/jetbrains/intellij/deps/intellij-coverage-agent/1.0.731/intellij-coverage-agent-1.0.731.jar">
          <sha256sum>b9cf4532f3f4ce9d723ecb123badd767cbcb97a83ddbee913979fff9ec311753</sha256sum>
        </artifact>
      </verification>
    </properties>
    <CLASSES>
      <root url="jar://$MAVEN_REPOSITORY$/org/jetbrains/intellij/deps/intellij-coverage-agent/1.0.731/intellij-coverage-agent-1.0.731.jar!/" />
    </CLASSES>
    <JAVADOC />
    <SOURCES>
      <root url="jar://$MAVEN_REPOSITORY$/org/jetbrains/intellij/deps/intellij-coverage-agent/1.0.731/intellij-coverage-agent-1.0.731-sources.jar!/" />
    </SOURCES>
  </library>
</component>