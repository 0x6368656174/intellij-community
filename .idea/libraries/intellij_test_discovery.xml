<component name="libraryTable">
  <library name="intellij-test-discovery" type="repository">
    <properties include-transitive-deps="false" maven-id="org.jetbrains.intellij.deps:intellij-test-discovery-agent:1.0.711" />
    <CLASSES>
      <root url="jar://$MAVEN_REPOSITORY$/org/jetbrains/intellij/deps/intellij-test-discovery-agent/1.0.711/intellij-test-discovery-agent-1.0.711.jar!/" />
    </CLASSES>
    <JAVADOC />
    <SOURCES />
  </library>
</component>