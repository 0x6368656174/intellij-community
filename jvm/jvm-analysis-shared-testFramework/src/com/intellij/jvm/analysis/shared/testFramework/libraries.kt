package com.intellij.jvm.analysis.shared.testFramework

import com.intellij.openapi.roots.ModifiableRootModel
import com.intellij.testFramework.PsiTestUtil
import com.intellij.util.PathUtil
import org.jetbrains.annotations.ApiStatus
import java.io.File

internal fun ModifiableRootModel.addJavaAnnotationsLibrary() {
  val annotationsJar = File(PathUtil.getJarPathForClass(ApiStatus::class.java))
  PsiTestUtil.addLibrary(this, "java-annotations", annotationsJar.parent, annotationsJar.name)
}