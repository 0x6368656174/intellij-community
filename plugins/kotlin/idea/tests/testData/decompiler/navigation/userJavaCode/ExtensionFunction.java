import testData.libraries.*;

class TestOverload {
    {
        Integer v = ExtensionFunctionKt.filter(5, x -> x % 2 == 1);
    }
}