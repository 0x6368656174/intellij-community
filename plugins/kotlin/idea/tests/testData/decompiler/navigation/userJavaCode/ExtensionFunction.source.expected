ExtensionFunctionKt.class
<2>// IntelliJ API Decompiler stub source generated from a class file
// Implementation of methods is not available

package testData.libraries

public inline fun <T> T.<3>filter(predicate: (T) -> kotlin.Boolean): T? { /* compiled code */ }
 Integer.class
public final class <1>Integer extends Number implements Comparable<Integer> {
